x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/field_stats_initializer.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiForm,
+  EuiFormRow,
+  EuiTitle,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { FC } from 'react';
+import React, { useMemo, useState, useCallback } from 'react';
+import { ENABLE_ESQL } from '@kbn/esql-utils';
+import type { AggregateQuery } from '@kbn/es-query';
+import { DataSourceTypeSelector } from './field_stats_initializer_view_type';
+import { useDataVisualizerKibana } from '../../../kibana_context';
+import { FieldStatsESQLEditor } from './field_stats_esql_editor';
+import type {
+  FieldStatisticsTableEmbeddableState,
+  FieldStatsInitialState,
+} from '../grid_embeddable/types';
+import { FieldStatsInitializerViewType } from '../grid_embeddable/types';
+import { isESQLQuery } from '../../search_strategy/requests/esql_utils';
+export interface FieldStatsInitializerProps {
+  initialInput?: Partial<FieldStatisticsTableEmbeddableState>;
+  onCreate: (props: FieldStatsInitialState) => void;
+  onCancel: () => void;
+  onPreview: (update: Partial<FieldStatsInitialState>) => Promise<void>;
+}
+
+const defaultESQLQuery = { esql: 'from kibana_sample_data_ecommerce | limit 10' };
```

Comment:
As discussed, we should copy what happens in the ES|QL panel type and use the first index alphabetically. Currently if you don't have that index, you will get an error if you add the panel using the non-existent index.